doc/deprecated.rst

```diff
@@ -188,6 +188,29 @@ Templates
   ``Template::loadTemplate()``); pass instances of ``Twig\TemplateWrapper``
   instead.
 
+* Having a "block" definition nested in another node that captures the output
```

Comment:
Not answering on the "why", i have'nt tried this PR locally yet... but i feel the before/after examples do not represent the same usage/need.

For instance (please don't be too harsh on this code 🙏 ) this is a pattern i used from time to time, and i _feel_ this is often how the set/block was used. 

https://github.com/symfony/ux/blob/8a66e74d5e7070e5cbf60042cae535fae28d3e7a/ux.symfony.com/templates/components/Code/CodeLine.html.twig

I don't mind we cannot in 4.0 and will find a better way, but the "after" here would not help in this situation, right ?